bindings/py/cpp_src/bindings/math/py_Random.cpp

```diff
@@ -63,16 +63,10 @@ namespace nupic_ext {
 
 
         Random.def("sample",
-            [](Random_t& r, py::array_t<nupic::UInt32>& population, py::array_t<nupic::UInt32>& choices)
+            [](Random_t& r, std::vector<nupic::UInt32>& population, nupic::UInt32 nChoices)
```

Comment:
@dkeeney why are we using py::array (-> numpy.array) and not std::vector (-> list[] in py) ? 
- This compiles, implementation is simpler
  - I could just adapt tests to pass

- can pybind even automatically wrap functions where possible? (so we wouldn't have to write this dummy wrapper)
  - from py, if you have numpy.array, you can easily get list, not the other way around. 

Do you agree to this modification (and possibly prefered way of writing the wrappers)?

Reply:
true, thanks! 
I'll revert this change for this PR. 
For discussion if we can find a simpler method than py::array: https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html